Sources/Atomic.swift

```diff
@@ -187,6 +181,15 @@ internal class Lock {
 			case EBUSY:
 				return false
 			default:
+				#if DEBUG && arch(arm)
```

Comment:
It looks like `EAGAIN` will be silently allowed when `DEBUG`, but will lead to an assertion failure when `!DEBUG`.

I wonder if it should be the other way around? I usually set `DEBUG` in builds where I want _more_ assertions.

Reply:
I guess at this point we should just treat `EAGAIN` as `EBUSY`. Just found this issue: https://github.com/ReactiveCocoa/ReactiveCocoa/issues/3598.

Reply:
The weirdest thing is that based on [the libpthread implementation](https://opensource.apple.com/source/libpthread/libpthread-301.30.1/src/pthread_mutex.c.auto.html) only the recursive lock can return `EAGAIN`, and it holds for implementations since El Capitan GM. All the reported issues do not seem to involve properties which are the only thing using recursive locks in ReactiveSwift.